src/vcpkg-test/manifests.cpp

```diff
@@ -1024,3 +1021,43 @@ TEST_CASE ("Serialize all the ports", "[all-manifests]")
         REQUIRE(*serialized_scf == scf);
     }
 }
+
+static Json::Object manifest_with_license(StringView license)
+{
+    Json::Object res;
+    res.insert("name", Json::Value::string("foo"));
+    res.insert("version", Json::Value::string("0"));
+    res.insert("license", Json::Value::string(license.to_string()));
+    return res;
+}
+
+static bool license_is_valid(StringView license, LicenseIdCheck check = LicenseIdCheck::Strict)
+{
+    return parse_spdx_license_expression(license, check).has_value();
+}
+
+TEST_CASE ("simple license in manifest", "[manifests][license]")
+{
+    auto manifest = manifest_with_license("");
+    manifest["license"] = Json::Value::null(nullptr);
+
+    REQUIRE(manifest_is_parseable(manifest));
+    REQUIRE_FALSE(manifest_is_parseable(manifest_with_license("")));
+    REQUIRE(manifest_is_parseable(manifest_with_license("MIT")));
+}
+
+TEST_CASE ("valid and invalid licenses", "[manifests][license]")
+{
+    REQUIRE(license_is_valid("mIt"));
```

Comment:
Only ~8 tests for a function with a cyclomatic complexity of like 40 seems insufficient?

At least there should be one test per explicitly named error kind. (As tests of `parse_spdx_license_expression` rather than manifest parsing)

Reply:
could you relook?